src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -46,11 +77,17 @@ public AuthorizeFilter(AuthorizationPolicy policy)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            var effectivePolicy = Policy ?? await AuthorizationPolicy.CombineAsync(PolicyProvider, AuthorizeData);
```

Comment:
Fair enough. I wasn't sure what the expectation was for the `IPolicyProvider`. Most Mvc provider types cache non-request specific things once and re-use the result, hence the question.